extmod/uasyncio/taskgroup.py

```diff
@@ -0,0 +1,244 @@
+# Adapted with permission from the EdgeDB project.
+# Adapted for MicroPython
+
+
+__all__ = ["TaskGroup"]
+
+from . import core
+from . import event
+from . import task
+
+DEBUG = False
+
+
+class TaskGroup:
+    def __init__(self):
+        self._entered = False
+        self._exiting = False
+        self._aborting = False
+        self._loop = None
+        self._parent_task = None
+        self._parent_cancel_requested = False
+        self._tasks = set()
+        self._errors = []
+        self._base_error = None
+        self._on_completed = None
+
+    def __repr__(self):
```

Comment:
Is it necessary to have a complicated repr like this?

Reply:
I'll reduce it to something simpler when _DEBUG is False.